src/api/procedures/moveFunds.ts

```diff
@@ -53,6 +72,21 @@ export async function checkArgs(
     });
   }
 
+  const checkAssetExists = proofs.map(({ asset }) => {
+    const confidentialAsset = asConfidentialAsset(asset, context);
+
+    return confidentialAsset.exists();
```

Comment:
(optional) I think it would be nice to report back the exact assets that don't exist in `data`. If the id gets grouped into the result it should be

```suggestion
    const exists = await confidentialAsset.exists()
    
    return { assetId: asset.id, exists }
```
    
Then
```ts
    const notFoundAssets = assetExists.filter(({ exists }) => exists)    
    if(notFoundAssets.length) {
    ...
    data: notFoundAssets.map(({ assetId }) => assetId)
 ```